### PR TITLE
feat(connection): show sources and sinks referencing the connection

### PR DESCRIPTION
**Pull request by WillyKidd**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- FEAT: in `SHOW CONNECTIONS`, display `sources` and `sinks` that reference the connection
- FIX: in 'DROP SINK', decrease connection ref if sink references connection
- TEST: add `e2e` tests for creating a sink with connection
- Example:
  image

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

### CODE REVIEW

**Review by StrikeW — COMMENTED** [review]
LGTM, please wait for the merge of https://github.com/risingwavelabs/risingwave/pull/9263 and add Sinks that used a connection.

**Review by StrikeW — APPROVED** [review]
LGTM

File: src/frontend/src/catalog/schema_catalog.rs [hunk 1]
```diff
@@ -52,6 +52,10 @@ pub struct SchemaCatalog {
     connection_by_name: HashMap<String, Arc<ConnectionCatalog>>,
     connection_by_id: HashMap<ConnectionId, Arc<ConnectionCatalog>>,
 
+    // This field is currently used only for `show connections`
+    connection_source_ref: HashMap<ConnectionId, Vec<SourceId>>,
```
Comment (StrikeW) [inline src/frontend/src/catalog/schema_catalog.rs:56]: Considering source and sink may not be dropped frequently, so it is ok to use a `Vec` to store those source ids.

**Comment by yezizp2012** [comment]
👍 LGTM. BTW it should be a `user-facing-changes` feature, please add some details in the documentation and release note part.

**Pull request merged**